dockstore-webservice/src/main/java/io/openapi/api/impl/ToolsApiServiceImpl.java

```diff
@@ -904,6 +924,11 @@ public static List<ToolFile> getToolFiles(Set<SourceFile> sourceFiles, List<Stri
             ToolFile toolFile = new ToolFile();
             toolFile.setPath(path.relativize(Paths.get(file.getAbsolutePath())).toString());
             ToolFile.FileTypeEnum fileTypeEnum = fileTypeToToolFileFileTypeEnum(file.getType());
+            // arbitrarily pick first checksum, seems like bug in specification, should probably be array
+            final List<Checksum> checksums = convertToTRSChecksums(file);
+            if (!checksums.isEmpty()) {
+                toolFile.setChecksum(convertToTRSChecksums(file).get(0));
```

Comment:
You already have `checksums`, you could do `toolFile.setChecksum(checksums.get(0))` instead of calculating it again.